### PR TITLE
Default collections and application delete

### PR DESCRIPTION
**Pull request by GERey**

### CODE REVIEW

File: stack/core/src/main/java/org/apache/usergrid/corepersistence/CpEntityManager.java [hunk 1]
```diff
@@ -745,7 +745,32 @@ public RelationManager getRelationManager( EntityRef entityRef ) {
     @Override
     public Set<String> getApplicationCollections() throws Exception {
 
-        return getRelationManager( getApplication() ).getCollections();
+        Set<String> existingCollections = getRelationManager( getApplication() ).getCollections();
+
+        //Handles reading custom collections.
+//        Set<String> dynamic_collections = cast( getDictionaryAsSet( getApplicationRef(), Schema.DICTIONARY_COLLECTIONS ) );
```
Comment (michaelarusso) [inline stack/core/src/main/java/org/apache/usergrid/corepersistence/CpEntityManager.java]: is this left here as a TODO?

File: stack/rest/src/main/java/org/apache/usergrid/rest/management/organizations/applications/ApplicationResource.java [hunk 2]
```diff
@@ -482,20 +482,27 @@ public ApiResponse executePut(  @Context UriInfo ui, String body,
...
 
-        Properties props = management.getProperties();
-
-        // for now, only works in test mode
-        String testProp = ( String ) props.get( "usergrid.test" );
-        if ( testProp == null || !Boolean.parseBoolean( testProp ) ) {
-            throw new UnsupportedRestOperationException("Test props not not functioning correctly.");
-        }
+//        Properties props = management.getProperties();
+//
```
Comment (michaelarusso) [inline stack/rest/src/main/java/org/apache/usergrid/rest/management/organizations/applications/ApplicationResource.java]: Can we remove the commented code?

File: stack/core/src/main/java/org/apache/usergrid/corepersistence/CpEntityManager.java (hunk 1 above)
Comment (GERey) [inline stack/core/src/main/java/org/apache/usergrid/corepersistence/CpEntityManager.java]: This was left here as a way to see how custom collections used to be queried but probably should be deleted and replaced with a jira ticket.

File: stack/rest/src/main/java/org/apache/usergrid/rest/management/organizations/applications/ApplicationResource.java (hunk 2 above)
Comment (GERey) [inline stack/rest/src/main/java/org/apache/usergrid/rest/management/organizations/applications/ApplicationResource.java]: This one was my bad. The commented out code should be good.

**Pull request merged**